### PR TITLE
(packaging) Add bootsnap dependency to bolt-runtime

### PR DESCRIPTION
**Pull request by jpogran**
Needed for https://github.com/puppetlabs/bolt/pull/2291

### CODE REVIEW

**Comment by jpogran** [comment]
target branch needs to be updated before merging

File: configs/components/rubygem-puppet.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 component 'rubygem-puppet' do |pkg, settings, platform|
-  pkg.version "6.19.1"
-  pkg.md5sum '8d68e677e6ba88e46eb00699dcd4930b'
+  pkg.version "7.0.0"
```
Comment (lucywyman) [inline configs/components/rubygem-puppet.rb:2]: I think we'll actually want to update to latest, 7.3.0. Does Bootsnap depend on this? It might be good to open a separate PR since I think Skeletor will have to compile Bolt with a different version of Puppet than Puppet 7.

Comment (jpogran) [reply configs/components/rubygem-puppet.rb:2]: I should have documented it in the commit message, my bad, but we need puppet 7.X because it removed the win32-dir gem. Bootsnap uses a hash lookup for finding cached file paths, and the gem folder name `win32-dir` does not match the gem require statement of `win32/dir`.

I'm fine with bumping to pupet 7.X in another PR before merging this.

**Comment by lucywyman** [comment]
Mmk - I needed to make a few updates to get this far, including updating one md5 sum and pinning msgpack back to 1.3.3 since 1.4.2 did not publish a mingw version of the gem: https://github.com/jpogran/puppet-runtime/pull/1

After that, I ran into this error:
```
touch rubygem-bootsnap-configure
touch rubygem-bootsnap-build
touch rubygem-bootsnap-check
export PATH="/cygdrive/c/ProgramData/chocolatey/lib/mingw/tools/install/mingw64/bin:/cygdrive/c/tools/pl-build-tools/bin:/cygdrive/c/tools/pl-build-tools/include:/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/bin:/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/bin:/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/include:/bin:/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/Packer/SysInternals:/cygdrive/c/Program Files/Git/cmd:/cygdrive/c/Program Files/PowerShell/7" && \
export CONFIGURE_ARGS="--with-cflags='-IC:/ProgramFiles64Folder/PuppetLabs/Bolt/include/ruby-2.5.0'" && \
export GEM_HOME="C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0" && \
cd ./ && \
C:/ProgramFiles64Folder/PuppetLabs/Bolt/bin/gem.bat install --no-document --local --bindir=C:/ProgramFiles64Folder/PuppetLabs/Bolt/bin bootsnap-1.7.2.gem
Building native extensions. This could take a while...
ERROR:  Error installing bootsnap-1.7.2.gem:
	ERROR: Failed to build gem native extension.

    current directory: C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap
C:/ProgramFiles64Folder/PuppetLabs/Bolt/bin/ruby.exe -I C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/2.7.0 -r ./siteconf20210302-2716-wmwd7j.rb extconf.rb
creating Makefile

current directory: C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap
make "DESTDIR=" clean
make[1]: Entering directory '/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap'
make[1]: Leaving directory '/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap'

current directory: C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap
make "DESTDIR="
make[1]: Entering directory '/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap'
generating bootsnap-x64-mingw32.def
compiling bootsnap.c
bootsnap.c:15:10: fatal error: ruby.h: No such file or directory
   15 | #include "ruby.h"
      |          ^~~~~~~~
compilation terminated.
Makefile:244: recipe for target 'bootsnap.o' failed
make[1]: *** [bootsnap.o] Error 1
make[1]: Leaving directory '/cygdrive/c/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2/ext/bootsnap'

make failed, exit code 2

Gem files will remain installed in C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/gems/bootsnap-1.7.2 for inspection.
Results logged to C:/ProgramFiles64Folder/PuppetLabs/Bolt/lib/ruby/gems/2.7.0/extensions/x64-mingw32/2.7.0/bootsnap-1.7.2/gem_make.out
Makefile:1011: recipe for target 'rubygem-bootsnap-install' failed
```

Which I suspect may be related to pinning msgpack back?

**Comment by jpogran** [comment]
I don't think so, each gem compile is isolated in that respect. The error means it can't find the c ruby header files, and that's why I modified line 82, they are in the include directory. If it's erroring, we'll have to run the build by hand using artifacts from a failed build and inspect the environment variables and paths. Not trivial.

Thanks for the hash updates, I had updated the versions but missed updating the hashes.

File: configs/platforms/windows-2012r2-x64.rb [hunk 2]
```diff
@@ -11,10 +11,10 @@
 
   # C:\tools is likely added by mingw, however because we also want to use that
   # dir for vsdevcmd.bat we create it for safety
-  plat.provision_with "mkdir C:/tools"
+  plat.provision_with "mkdir -p C:/tools"
   # We don't want to install any packages from the chocolatey repo by accident
   plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe upgrade -y chocolatey"
-  plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe sources remove -name chocolatey"
+  # plat.provision_with "C:/ProgramData/chocolatey/bin/choco.exe sources remove -name chocolatey"
```
Comment (gimmyxd) [inline configs/platforms/windows-2012r2-x64.rb:17]: @jpogran  I think that by removing this, a newer `cmake` version is installed now(3.19.6, previously it was 3.5.2). This is breaking yaml-cpp build on agent-runtime. Would pining `cmake` to `3.5.2` work for bolt?

Comment (jpogran) [reply configs/platforms/windows-2012r2-x64.rb:17]: Apologies, this was my mistake. I was supposed to go back to using internal repo only. Will post a fix shortly
